debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -176,48 +177,75 @@ private void takeSnapshot(Consumer<ChangeEvent> consumer) {
             logger.info("Step 3: reading and exporting the contents of each table");
             AtomicInteger rowsCounter = new AtomicInteger(0);
             final Map<TableId, String> selectOverrides = getSnapshotSelectOverridesByTable();
-            schema.tables().forEach(tableId -> {
+
+            // whether this thread has been interrupted (i.e. the connector has been stopped before the snapshot was completed)
+            boolean interrupted = false;
```

Comment:
IMHO this is superfluous and just duplicates/simulates thred *interrupted* status. The catch blocks should look like
```java
 catch (InterruptedException e) {
      Thread.interrupt(); // Note this is different method, it sets back again the interrupted status
}
```
And the conditions should call ```Thread.isInterrupted()```

Reply:
Hum, I don't feel comfortable about relying on the thread's interrupt status, as it might be altered behind our backs. The pattern here is also what we do for the MySQL connector. I'll take another look tomorrow.

Reply:
Yes, I think you are right, this should be done differently. I've pushed an update, there's just a single catch for the `InterruptedException` now, that simplifies that quite a bit.